### PR TITLE
Integrate Nix into the project and remove Conan from Cmake

### PR DESCRIPTION
**Pull request by ChronosXYZ**
Now the repo is just CMake header-only library. Integrating Conan tightly with CMake build script is pretty bad idea. Also this repo was integrated with old version of Conan, so this adds security issues. I've added Nix flake to easily have proper environment with necessary dependencies and compilers. You can read about Nix here: https://nixos.org/

### CODE REVIEW

**Comment by Hual** [comment]
>Integrating Conan tightly with CMake build script is pretty bad idea

why?

**Comment by ChronosXYZ** [comment]
1. Security vulnerabilities because of old version of Conan
2. This breaks reproducibility of the build in some cases. IMHO CMake should only search for libraries, and external package manager, which is conan, vcpkg, nix, whatever, should install libraries.

**Comment by Hual** [comment]
> 1. Security vulnerabilities because of old version of Conan
This should be handled by updating conan support, not removing it completely

> 2. This breaks reproducibility of the build in some cases
I don't think that's true in this specific case since we're using a specific version of header-only libraries

> IMHO CMake should only search for libraries
This sounds great in theory, doesn't work well in practice; that was the whole point of having a single compiler and a single source of dependencies - fewer differences between systems

> or in another way, why main repo includes SDK?
It was easier to maintain a rapidly developing SDK as part of the main project instead of having to sync a submodule; nowadays it's stable so it might not be needed anymore

Bottom line is I don't like having even more cmake code for SDK; I think git submodules would be a better idea at this stage (@AmyrAhmady agrees)

**Comment by ChronosXYZ** [comment]
> > 1. Security vulnerabilities because of old version of Conan
> 
> This should be handled by updating conan support, not removing it completely
> 
> > 2. This breaks reproducibility of the build in some cases
> 
> I don't think that's true in this specific case since we're using a specific version of header-only libraries
> 
> > IMHO CMake should only search for libraries
> 
> This sounds great in theory, doesn't work well in practice; that was the whole point of having a single compiler and a single source of dependencies - fewer differences between systems
> 
> > or in another way, why main repo includes SDK?
> 
> It was easier to maintain a rapidly developing SDK as part of the main project instead of having to sync a submodule; nowadays it's stable so it might not be needed anymore
> 
> Bottom line is I don't like having even more cmake code for SDK; I think git submodules would be a better idea at this stage (@AmyrAhmady agrees)

sure, but i still would like to have Nix support for the SDK. I will try to make it with Conan
